libr/util/name.c

```diff
@@ -40,12 +45,35 @@ R_API int r_name_filter(char *name, int maxlen) {
 			*name = '\0';
 			break;
 		}
-		if (!r_name_validate_char (*name)) {
+		if (!r_name_validate_char (*name) && *name != '\\') {
 			*name = '_';
 			//		r_str_ccpy (name, name+1, 0);
 			//name--;
 		}
 	}
+	while (i > 0) {
+		if (*(name - 1) == '\\' && is_special_char (name)) {
+			*name = '_';
+			*(name - 1) = '_';
+		}
+		if (*name == '\\') {
+			*name = '_';
+		}
+		name--;
+		i--;
+	}
+	if (*name == '\\') {
+		*name = '_';
+	}
+	// trimming trailing and leading underscores
+	len = strlen (name);
+	for (len; *(name + len - 1) == '_'; len--);
```

Comment:
Just use
```
for (;
```
That non-initialization of `len` can be misread as a check of `len` against 0.